doc/swish/erlang.tex

```diff
@@ -758,6 +760,128 @@ \subsection{Miscellaneous}
+In either case, it adds \var{r} to the weak eq-hashtable
+consulted by \code{foreign-handle-count}, etc.
+To ensure that \var{r} is registered with the guardian just once,
+\var{r} should already contain \var{handle} the first time
+the procedure is called with \var{r} and a \var{handle} that
+is not \code{\#f}.
+
+% ----------------------------------------------------------------------------
+\defineentry{count-foreign-handles}
+\begin{procedure}
+  \code{(count-foreign-handles \var{obj} \var{report-count})}
+\end{procedure}
+\returns{} \var{obj}
+
+The \code{count-foreign-handles} procedure takes an arbitrary \var{obj} and a
+procedure \var{report-count} that takes \var{obj}, a symbol \var{type}
+identifying a source of foreign handles, and \var{count}, the number of
+open handles of that type.
+The \code{count-foreign-handles} procedure calls \var{report-count} once for
+each foreign-handle type registered with \code{make-foreign-handle-guardian},
+in an unspecified order.
+
+% ----------------------------------------------------------------------------
+\defineentry{foreign-handle-count}
+\begin{procedure}
+  \code{(foreign-handle-count \var{type-name})}
+\end{procedure}
+\returns{} a procedure
+
+The \code{foreign-handle-count} procedure takes a symbol \var{type-name} that
+must already have been registered with \code{make-foreign-handle-guardian}
+and returns a procedure of no arguments that returns the number of open foreign
+handles of that type.
+
+% ----------------------------------------------------------------------------
+\defineentry{foreign-handle-print}
+\begin{procedure}
+  \code{(foreign-handle-print \var{type-name})}
+\end{procedure}
+\returns{} a procedure
+
+The \code{foreign-handle-print} procedure takes a symbol \var{type-name} that
+must already have been registered with \code{make-foreign-handle-guardian} and
+returns a procedure that prints information about open handles of that type.
+The procedure returned uses the \var{print} and \var{get-create-time}
+procedures registered with \code{make-foreign-handle-guardian}.
+The procedure takes an optional textual output port \var{op}, which defaults
+to the current output port, and calls \var{print} for each open handle of the
+designated type in increasing order of the creation time from \var{get-create-time}
+and then by increasing handle.
+
+% ----------------------------------------------------------------------------
+\defineentry{print-foreign-handles}
+\begin{procedure}
+  \code{(print-foreign-handles \opt{\var{op}})}
+\end{procedure}
+\returns{} unspecified
+
+The \code{print-foreign-handles} procedure prints information about open
+foreign handles to textual output port \var{op}, which defaults to the current
+output port.
+The \code{print-foreign-handles} procedure
+calls the procedure returned by \code{(foreign-handle-print\ \var{type-name})}
+for each foreign-handle type registered with \code{make-foreign-handle-guardian}.
+
```

Comment:
In arg-check below, the \ldots are indented too far. They should be at the name level as the let of arg.

Reply:
Good catch. Also fixed it so that `(arg-check 'foo)` doesn't result in a syntax error from `let`. Derived syntax might expand into an `arg-check` with no arguments.